cmd/kinflate/demo/springboot.md

```diff
@@ -0,0 +1,328 @@
+spring.datasource.url=jdbc:mysql://<prod_database_host>:3306/db_example
+spring.datasource.username=root
+spring.datasource.password=admin
+EOF
+
+kinflate add configmap demo-configmap --from-file application-prod.properties
+
+cat Kube-manifest.yaml
+```
+`Kube-manifest.yaml`'s configMapGenerator section should contain:
+> ```
+> configMapGenerator:
+> - files:
+>   - application.properties
+>   - application-prod.properties
+>   name: demo-configmap
+> kind: Manifest
+> ```
+
+### Name Customization
+
+Arrange for the resources to begin with prefix
+_prod-_ (since they are meant for the _production_
+environment):
+
+<!-- @customizeLabel @test -->
+```
+cd $DEMO_HOME
+
+kinflate set nameprefix 'prod-'
+
+cat Kube-manifest.yaml
+```
+
+`Kube-manifest.yaml` should have updated value of namePrefix field:
+
+> ```
+> apiVersion: manifest.k8s.io/v1alpha1
+> ....
+> namePrefix: prod-
+> objectAnnotations:
+>  note: This is a example annotation
+> ```
+
+This `namePrefix` directive adds _prod-_ to all
+resource names.
+
+<!-- @genNamePrefixConfig @test -->
+```
+kinflate inflate $DEMO_HOME
+```
+
+The output should contain:
+> ```
+> apiVersion: v1
+> data:
+>   ....
+> kind: ConfigMap
+> metadata:
+>   ....
+>   name: prod-demo-configmap-7746248cmc
+> ---
+> apiVersion: v1
+> kind: Service
+> metadata:
+>   ....
+>   name: prod-sbdemo
+> spec:
+>   ....
+> ---
+> apiVersion: apps/v1beta2
+> kind: Deployment
+> metadata:
+>   ....
+>   name: prod-sbdemo
+> spec:
+>   selector:
+>     ....
+> ```
+
+### Label Customization
+
+We want resources in production environment to have
+certain labels so that we can query them by label
+selector.
+
+`kinflate` does not have `set label` command to add
+label, but we can edit `Kube-manifest.yaml` file under
+`prod` directory and add the production labels under
+`objectLabels` fields as highlighted below.
+
+<!-- @customizeLabels @test -->
+```
+sed -i 's/app: helloworld/app: prod/' \
+    $DEMO_HOME/Kube-manifest.yaml
+```
+
+At this point, running `kinflate inflate` will
+generate MySQL configs with name-prefix 'prod-' and
+labels `env:prod`.
+
+
+### Download Patch for JVM memory set and health check
+When a Spring Boot application is deployed in a k8s cluster, the JVM is running inside a container. We want to set memory limit for the container and make sure
+the JVM is aware of that limit. In K8s deployment, we can set the resource limits for containers and inject these limits to 
+some environment variables by downward API. When the container starts to run, it can pick up the environment variables and
+set JVM options accordingly.
+
+We also want to add liveness check and readiness check in the production environment. Spring Boot application
+has end points such as `/actuator/health` for this. We can customize the k8s deployment resource to talk to Spring Boot end point.
+
+Download the patch `patch.yaml`. It contains the memory limits setup and livenessProbes.
+<!-- @downloadPatch @test -->
```

Comment:
was wondering if separating out these patches is better idea. Not a deal breaker, but something to think.

Having each patch do one thing is a good practice to communicate. That way people can turn on/off individual patches (mix-ins kinda behavior).

Reply:
It is better. I was thinking the same when I wrote the demo. The reason they were put together is that the file is downloaded from spring boot instance repo, where it is as a single patch file. 
I will separate the patches in the repo and update here.